examples/api/test/widgets/scroll_view/custom_scroll_view.1_test.dart

```diff
@@ -0,0 +1,216 @@
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    expect(
+      find.byType(Scaffold),
+      findsOne,
+      reason: 'Expected to have a Scaffold in the App',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(Scaffold),
+        matching: find.byType(AppBar),
+      ),
+      findsOne,
+      reason: 'Expected to have an Appbar in the Scaffold',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(AppBar),
+        matching: find.byType(IconButton),
+      ),
+      findsOne,
+      reason: 'Expected an IconButton in the AppBar',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(Scaffold),
+        matching: find.byType(CustomScrollView),
+      ),
+      findsOne,
+      reason: 'Expected a CustomScrollView in the Scaffold',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.byType(SliverList),
+      ),
+      findsOne,
+      reason: 'Expected one, initial (bottom) SliverList in the CustomScrollView',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 0'),
+      ),
+      findsOne,
+      reason: 'Expected element with text "Item: 0" in the CustomScrollView',
+    );
+  });
+
+  testWidgets('IconButton click extends existing SliverList', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.byType(SliverList),
+      ),
+      findsOne,
+      reason: 'Expected one, initial (bottom) SliverList in the CustomScrollView',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(SliverList),
+        matching: find.byType(Container),
+      ),
+      findsOne,
+      reason: 'Expected one, initial (bottom) Container in the SliverList',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 1'),
+      ),
+      findsNothing,
+      reason: 'Expected no element with text "Item: 1" in the CustomScrollView',
+    );
+
+    // Tap the IconButton in the AppBar
+    final Finder iconButtonFinder = find.descendant(
+      of: find.byType(AppBar),
+      matching: find.byType(IconButton),
+    );
+    await tester.tap(iconButtonFinder);
+    await tester.pump();
+
+    expect(
+      find.descendant(
+        of: find.byType(SliverList),
+        matching: find.byType(Container),
+      ),
+      findsExactly(2),
+      reason: 'Expected additional Container in the SliverList after the IconButton click',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 1'),
+      ),
+      findsOne,
+      reason: 'Expected element with text "Item: 1" in the CustomScrollView',
+    );
```

Comment:
```suggestion
    expect(find.widgetWithText(CustomScrollView, 'Item: 1'));
```

`widgetWithText` is a nice utility that helps to reduce code (and improve readability).
Consider applying it to all similar cases in this file.